### DISCUSSION TITLE
GitHub Actions downloads (probably) cached files

### DISCUSSION THREAD

**Discussion by Slluxx** [community/community#25741] [Actions]
<p>I have edited this post after tinkering a little bit.</p>
<p>This is a test file i created:</p>
<pre><code class="lang-auto">name: tests
on:
  workflow_dispatch:
jobs:
  tests:
    runs-on: ubuntu-latest
    steps:
      - name: Set up python 3.x
        uses: actions/setup-python@v2
        with:
          python-version: '3.x' # Version range or exact version of a Python version to use, using SemVer's version range syntax
          architecture: 'x64' # optional x64 or x86. Defaults to x64 if not specified
      - run: pip install requests
      - run: python -c "import requests as A;import json;print(json.loads(A.get('https://www.pokemon.com/api/pokemontv/v2/channels/us/').content)[0])"
</code></pre>
<p>The last command prints always <code>{'channel_description': 'Ash and Pikachu’s exciting adventu</code> …<br>
When the URL is openend inside a browser or if the code is executed through python on my local pc, it always prints a newer version of the json, <strong>not containing</strong> what is printed through the GH Action.<br>
So this is not a case of the json having items in a random order.</p>
<p>When working with APIs, id like to get a non-cached and up to date file. Why do i get an old one and how can i force a fresh one? Appending a timestamp as get parameter to make the url unique doesnt work.</p>

**Comment by airtower-luna** [comment]
<p>Exactly how do you download the files? You mention the Python <code>requests</code> module and headers in the post, but how do you create the URLs?</p>

**Comment by Slluxx** [comment]
<pre><code class="lang-python">with open('mappings.json') as json_file:
    mappings = json.load(json_file)
    for lang in mappings:
        currTime = str(int(time.time()))
        try:
            r = requests.get(mappings[lang]['apiUrl'] + "?time=" + currTime)
            rContent = r.content
        except:
            print("Error getting " + mappings[lang]['apiUrl'])
            continue
        try:
            a_json = json.loads(rContent)
            with open('./apiData/' + lang + '.json', 'wb') as f:
                f.write(rContent)

                if lang == "us":
                    print(a_json[0])
        except:
            print(mappings[lang]['apiUrl'])
            print("Could not be converted to JSON")
</code></pre>
<p>but how i “create” the url shouldnt be of importance right?</p>
<p>with this code ive even tried to add a timestamp at the end of the url to make it unique but that didnt work either. still getting the old file. and i am sure its downloaded because of the <code>print(a_json[0])</code> line.</p>

**Comment by Slluxx** [comment]
<pre><code class="lang-auto">name: tests
on:
  workflow_dispatch:
jobs:
  tests:
    runs-on: ubuntu-latest
    steps:
      - name: Set up python 3.x
        uses: actions/setup-python@v2
        with:
          python-version: '3.x' # Version range or exact version of a Python version to use, using SemVer's version range syntax
          architecture: 'x64' # optional x64 or x86. Defaults to x64 if not specified
      - run: pip install requests
      - run: python -c "import requests as A;import json;print(json.loads(A.get('https://www.pokemon.com/api/pokemontv/v2/channels/us/').content)[0])"
</code></pre>
<p>in my workflow, this prints <code>{'channel_description': 'Ash and Pikachu’s exciting adventu</code> …<br>
opening the url in the browser and searching for that string reveals that it doesnt exist</p>

**Comment by Slluxx** [comment]
<p>i have edited the title and first post to better reflect my issue</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="Slluxx" data-post="3" data-topic="222609">
<div class="title">
<div class="quote-controls"></div>
 Slluxx:</div>
<blockquote>
<p>but how i “create” the url shouldnt be of importance right?</p>
</blockquote>
</aside>
<p>I meant the exact URL might matter, but with the one in the last example I don’t see a potential problem. There could be differences on the server side though. For example, there is a <code>country_codes</code> element in the response, so maybe there are different answers based on guessed location or something?</p>

**Comment by Slluxx** [comment]
<p>edit: nevermind. you are right</p>
<p>ive tested with a vpn again and i am getting different data based on the country i am in.<br>
this makes no sense though. why would anyone have an api with /us, /de, etc endpoints and still deliver content based on the region?</p>
